pkg/database/database.go

```diff
@@ -137,53 +139,90 @@ func OpenDb(op *DbOptions, systemDB DB, log logger.Logger) (DB, error) {
 		return dbi, nil
 	}
 
-	err = dbi.sqlEngine.UseDatabase(dbInstanceName)
+	dbi.sqlInitCancel = make(chan struct{})
+	dbi.sqlInit.Add(1)
+
+	go func() {
+		defer dbi.sqlInit.Done()
+
+		dbi.Logger.Infof("Loading SQL Engine for database '%s' (replica = %v)...", op.dbName, op.replicationOpts.Replica)
+
+		err := dbi.initSQLEngine(systemDB)
+		if err != nil {
+			dbi.Logger.Errorf("Unable to load SQL Engine for database '%s' (replica = %v). %v", op.dbName, op.replicationOpts.Replica, err)
+			return
```

Comment:
What happens here? Is the database in an always-broken state until restart?

Reply:
currently, if there is an error while initiating sql engine yes, there is no re-attempt